### ISSUE TITLE
Manage multiple events on save pipeline (PART II)

### ISSUE THREAD

**Issue by zeppaman** [arduosoft/RawCMS#166]
On same branch of Manage multiple events on save pipeline (PART II).

# Adapt presave lambda
Events are managed by one lambda called `JsDispatcher`. The relevant code is:

```cs
public override void Execute(string collection, ref JObject item, ref Dictionary<string, object> dataContext)
        {
            var settings = this.entityService.GetByName(collection);
            if (settings != null)
            {
                if (!string.IsNullOrEmpty(settings.PresaveScript))
                {
                    Dictionary<string, object> input = item.ToObject<Dictionary<string, object>>();
                    Engine add = new Engine()
                      .SetValue("item", input)
                     .Execute(settings.PresaveScript);
                    item = JObject.FromObject(input);
                }
            }
        }
```
*What to change here*
1. Make it abstract and let inherit from DataProcessLambda (this will let be instanced to the specific scope)
2.  the Execute method has to be renamed to ExecuteInternal. This part has to be changed to adapt to the new model (i.e. script is on settings.Events.PreSave). Basing on input (a string with the event name that will match the element inside Events)
This class will be the base for all implementation

# Manage all other events
Basically you have to replicate the same for all other events. You have to create a class for each event XXXJsDispatcher where XXX is PreSave, PostSave, PreDelete, Post Delete. All of these classes will implement JsDispatcher. In each class will specify the stage (pre or post) and the operation (write or delete). 

# References
The full-text plugin work using the same pattern see https://github.com/arduosoft/RawCMS/blob/master/Plugins/RawCMS.Plugins.FullText/Lambdas/FullTextLambda.cs
So, in your case, you can have one file with the base class (JsDispatcher) and the 4 implementations specialized on each context
